### PR TITLE
Battery: add option to send resting voltage estimate to GCS

### PR DESCRIPTION
**Pull request by tridge**
This is what I want to see on my GCS when I'm flying a quadplane, it is much more useful than current voltage

### CODE REVIEW

**Comment by IamPete1** [comment]
We do have a couple of related PRs, https://github.com/ArduPilot/ardupilot/pull/17870 https://github.com/ArduPilot/ardupilot/pull/17869. Would be nice to have both resting and true at once.

**Comment by tridge** [comment]
> Would be nice to have both resting and true at once.

yes, it would be, and I think that effort is complementary to this PR. This PR works with existing ground stations. 
In practice I really only want to see one voltage displayed on my GCS, and that is the resting voltage. Showing both would just be too much clutter on the display.
The main advantage of transmitting both is it allows it to be a GCS option which one to display, which is preferable to an option on the flight controller.

**Comment by tridge** [comment]
this was tested very successfully on 3 aircraft today

File: libraries/AP_BattMonitor/AP_BattMonitor.cpp [hunk 1]
```diff
@@ -438,6 +438,19 @@ float AP_BattMonitor::voltage_resting_estimate(uint8_t instance) const
...
+float AP_BattMonitor::gcs_voltage(uint8_t instance) const
+{
+    if ((_params[instance]._options.get() & uint32_t(AP_BattMonitor_Params::Options::GCS_Resting_Voltage)) != 0) {
+        return voltage_resting_estimate(instance);
+    }
+    if (instance < _num_instances) {
+        return state[instance].voltage;
+    } else {
+        return 0.0f;
+    }
```
Comment (peterbarker) [inline libraries/AP_BattMonitor/AP_BattMonitor.cpp:451]:
```suggestion
    }
    return 0.0f;
```

**Review by peterbarker — APPROVED** [review]

**Review by hendjoshsr71 — APPROVED** [review]

**Pull request merged**